src/main/java/nz/cri/gns/NZSHM22/util/DataLogger.java

```diff
@@ -0,0 +1,192 @@
+package nz.cri.gns.NZSHM22.util;
+
+import cern.colt.matrix.tdouble.DoubleMatrix2D;
+import java.io.*;
+import java.nio.file.Files;
+import java.text.DecimalFormat;
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+public class DataLogger {
+
+    static DecimalFormat fmt = new DecimalFormat("0.###E0");
+
+    /**
+     * Takes a double value and returns the smallest ASCII representation with 3 decimal digit
+     * accuracy.
+     *
+     * @param value
+     * @return
+     */
+    public static String format(double value) {
+
+        if (value == 0) {
+            return "0";
+        }
+
+        String a = value + "";
+
+        if (Double.isInfinite(value) || Double.isNaN(value)) {
+            return a;
```

Comment:
should we take this opportunity to not name variables with a single character?

Reply:
done